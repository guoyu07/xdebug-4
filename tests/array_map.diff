004- %w%f %w%d       -> strlen('a') /%s/array_map.php:0
005- %w%f %w%d       -> strlen('bb') /%s/array_map.php:0
006- %w%f %w%d       -> strlen('ccc') /%s/array_map.php:0