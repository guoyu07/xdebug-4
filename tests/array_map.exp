array
TRACE START [%d-%d-%d %d:%d:%d]
%w%f %w%d     -> array_map('strlen', array (0 => 'a', 1 => 'bb', 2 => 'ccc')) /%s/array_map.php:5
%w%f %w%d       -> strlen('a') /%s/array_map.php:0
%w%f %w%d       -> strlen('bb') /%s/array_map.php:0
%w%f %w%d       -> strlen('ccc') /%s/array_map.php:0
%w%f %w%d     -> gettype(array (0 => 1, 1 => 2, 2 => 3)) /%s/array_map.php:7
%w%f %w%d     -> file_get_contents('/tmp/%s') /%s/array_map.php:9