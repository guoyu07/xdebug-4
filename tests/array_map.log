
---- EXPECTED OUTPUT
array
TRACE START [%d-%d-%d %d:%d:%d]
%w%f %w%d     -> array_map('strlen', array (0 => 'a', 1 => 'bb', 2 => 'ccc')) /%s/array_map.php:5
%w%f %w%d       -> strlen('a') /%s/array_map.php:0
%w%f %w%d       -> strlen('bb') /%s/array_map.php:0
%w%f %w%d       -> strlen('ccc') /%s/array_map.php:0
%w%f %w%d     -> gettype(array (0 => 1, 1 => 2, 2 => 3)) /%s/array_map.php:7
%w%f %w%d     -> file_get_contents('/tmp/%s') /%s/array_map.php:9
---- ACTUAL OUTPUT
array
TRACE START [2010-04-16 12:23:42]
    0.0155     626864     -> array_map('strlen', array (0 => 'a', 1 => 'bb', 2 => 'ccc')) /Users/johann/src/xdebug/tests/array_map.php:5
    0.0156     627504     -> gettype(array (0 => 1, 1 => 2, 2 => 3)) /Users/johann/src/xdebug/tests/array_map.php:7
    0.0157     627504     -> file_get_contents('/tmp/xdt4bc856ce65e983.20934578.xt') /Users/johann/src/xdebug/tests/array_map.php:9
---- FAILED
