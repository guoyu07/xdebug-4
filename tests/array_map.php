<?php
$tf = xdebug_start_trace('/tmp/'. uniqid('xdt', TRUE));

$ar = array('a', 'bb', 'ccc');
$r = array_map('strlen', $ar);

echo gettype($r), "\n";

echo file_get_contents($tf);
unlink($tf);
?>
