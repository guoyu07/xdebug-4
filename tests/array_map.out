array
TRACE START [2010-04-16 12:23:42]
    0.0155     626864     -> array_map('strlen', array (0 => 'a', 1 => 'bb', 2 => 'ccc')) /Users/johann/src/xdebug/tests/array_map.php:5
    0.0156     627504     -> gettype(array (0 => 1, 1 => 2, 2 => 3)) /Users/johann/src/xdebug/tests/array_map.php:7
    0.0157     627504     -> file_get_contents('/tmp/xdt4bc856ce65e983.20934578.xt') /Users/johann/src/xdebug/tests/array_map.php:9